src/main/java/org/json/JSONArray.java

```diff
@@ -1500,6 +1519,44 @@ public Writer write(Writer writer, int indentFactor, int indent)
     public boolean isEmpty() {
         return this.myArrayList.isEmpty();
     }
+
+
+    /**
+     * Add a collection's elements to the JSONArray.
+     *
+     * @param collection
+     *            A Collection.
+     */
+    private void addAll(Collection<?> collection) {
+        for (Object o: collection){
```

Comment:
Add a new line:
```java
this.myArrayList.ensureCapacity(this.myArrayList.size()+ collection.size());
```

Reply:
Implemented in 86e136a

Reply:
I think you missed this one. You got the one for the `addAll(Array)`, but not for `addAll(Collection)`